nncf/dynamic_graph/context.py

```diff
@@ -425,9 +429,14 @@ def in_operator(self, val):
         self._thread_local.in_operator = val
 
     @property
-    def scope_modules(self):
+    def module_call_stack(self) -> List[torch.nn.Module]:
         self._init_thread_local()
-        return self._thread_local.scope_modules
+        return self._thread_local.module_call_stack
+
+    def get_current_module(self) -> Optional[torch.nn.Module]:
+        if self.module_call_stack:
+            return self.module_call_stack[-1]
```

Comment:
any issues with multi-threading? I wonder how reliable this approach to get current module.

Reply:
The access is being done to a thread-local variable, as is evident from the code somewhat below these lines. Also notice that the existing code uses scopes in order to launch operator hooks, and it gets said scopes directly from the module object class names, so basically if it works for module class names, it also works for the module objects themselves.